creativecommons/cccatalog-frontend
# Create separate sessions for a/b testing and usage analytics

## Fixes


Fixes #1236 by @aldenstpage 

## Description

A new session Id is created for every session. This way, a new session will be recorded by analytics  each time a user opens CC Search.

## Technical details

I moved the logic of session Id creation and saving it to a session cookie to 'utils' folder. It is used both for a/b testing and analytics session Id creation. `user` store contains `usageSessionId` and `abSessionId`. 

If we save the session Id in a session cookie without expiration date, it will not be deleted if a user has set "Continue where you left off" (Chrome) or "Restore previous session" (Firefox) on browser closing. To reset the session Id for each session, I set the expiry date to 1 hour (or, currently, to 1 minute for testing purposes).

## Tests


1. Go to search page. Type something in the search input and click enter.
2. Open 'Cookies' in the 'Application' Tab in the dev tools.
3. You can see two cookies: abSessionId and 'usageSessionId'.
4. The 'usageSessionId' is persisted until the user closes or refreshes the tab. This cookie is sent to Analytics API.

## Checklist
 
  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
